cmd/find-main.go

```diff
@@ -0,0 +1,141 @@
+/*
+ * Minio Client (C) 2014, 2015 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/minio/cli"
+	"github.com/minio/mc/pkg/console"
+)
+
+// find specific flags
+var (
+	findFlags = []cli.Flag{
+		cli.StringFlag{
+			Name:  "name",
+			Usage: "Find files of a matching name",
+		},
+		cli.StringFlag{
+			Name:  "path",
+			Usage: "Find global matches on the entire path",
+		},
+		cli.StringFlag{
+			Name:  "regex",
+			Usage: "Matches all files with PCRE regular expression",
+		},
+		cli.StringFlag{
+			Name:  "print",
+			Usage: "Prints the found files given args to STDOUT",
+		},
+		cli.StringFlag{
+			Name:  "mcexec",
```

Comment:
Also NOTE: this is not decided yet from the command point of view until @abperiasamy provides his views - It was my *two cents* to avoid the complexity of handling all kinds of use cases and defining multiple `exec` styles.